### PR TITLE
AI Mappings

### PR DESCRIPTION
**Pull request by Sollace**
Mapped the new classes used by goats, and performed some routine maintenance on all of the AI classes.

I meticulously went through every AI class and made sure they're fully mapped. No class name changes, just minor stuff and fixes. :tiny_potato:

### CODE REVIEW

File: mappings/net/minecraft/entity/ai/brain/Activity.mapping
Comment (OroArmor) [inline mappings/net/minecraft/entity/ai/brain/Activity.mapping:7]: This isnt really needed, or use `other`

**Review by OroArmor — CHANGES_REQUESTED** [review]
Looks good except `EXPIRY`. I know you are following what is there, but its not immediately clear what it is.

File: mappings/net/minecraft/entity/LivingEntity.mapping
Comment (liach) [inline mappings/net/minecraft/entity/LivingEntity.mapping]: Sure, but I hope you can document that when an entity is "leaping", it does not experience drag (on its speed) horizontally

File: mappings/net/minecraft/entity/ai/brain/Activity.mapping
Comment (liach) [reply mappings/net/minecraft/entity/ai/brain/Activity.mapping:7]: `o` is the name from `java.lang.Object`'s source code. I support sollace.

File: mappings/net/minecraft/entity/ai/brain/task/WalkTask.mapping
Comment (liach) [inline mappings/net/minecraft/entity/ai/brain/task/WalkTask.mapping]:
```suggestion
	FIELD field_30111 speed F
	METHOD <init> (F)V
		ARG 1 speed
```
Assuming there is no name conflict.

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/entity/LivingEntity.mapping
Comment (liach) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: I mean, this api is very likely used by mods, so documenting this would definitely help a lot.

File: mappings/net/minecraft/entity/ai/brain/Activity.mapping
Comment (enbrain) [reply mappings/net/minecraft/entity/ai/brain/Activity.mapping:7]: It's okay to use `o` in this PR imo. We need a separate PR for refactoring this thing anyway (#2299).

**Review by OroArmor — APPROVED** [review]

File: mappings/net/minecraft/entity/LivingEntity.mapping
Comment (Sollace) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: Fair point! In which case calling it "leaping" may not be the correct name to use (I thought it odd that this isn't in the goat class).

File: mappings/net/minecraft/entity/ai/goal/IronGolemWanderAroundGoal.mapping
Comment (YanisBft) [inline mappings/net/minecraft/entity/ai/goal/IronGolemWanderAroundGoal.mapping]:
```suggestion
	FIELD field_30214 ENTITY_COLLISION_RANGE I
```

File: mappings/net/minecraft/entity/ai/pathing/LandPathNodeMaker.mapping
Comment (YanisBft) [inline mappings/net/minecraft/entity/ai/pathing/LandPathNodeMaker.mapping]:
```suggestion
	METHOD method_29304 checkBoxCollision (Lnet/minecraft/class_238;)Z
```

**Review by YanisBft — APPROVED** [review]
Good to merge once conflicts are gone

**Comment by Sollace** [comment]
@YanisBft Conflics are fixed, but please don't merge yet. I have some revisions to make.

**Comment by Sollace** [comment]
@YanisBft All done. Feel free to review the new changes (there's very few). These are to address previous comments form earlier review.

File: mappings/net/minecraft/entity/LivingEntity.mapping
Comment (Gegy) [inline mappings/net/minecraft/entity/LivingEntity.mapping]: This seems a bit misleading- is this not more specifically related to having no friction?

Comment (Sollace) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: The way it's implemented doesn't seem like it really cares whether the entity is on the ground or not. Someone else is welcome to double-check that, though.

Comment (Gegy) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: There is still friction from air when off the ground, which this disables

Comment (Sollace) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: So you think change it to `noFriction` or `frictionless`? If you make a suggestion I'll happily accept.

Comment (liach) [reply mappings/net/minecraft/entity/LivingEntity.mapping]:
```suggestion
	FIELD field_30082 noDrag Z
```
This is drag. Drag is like multiplied onto velocity as a negative penalty.

Inertia is the tendency to keep the current velocity. You naming is exactly wrong because if there is no inertia, the entity should lose all velocity in the next tick.

Friction is a force that's applied when touching a surface. This is not friction either.

So `noDrag`. There is nothing resembling real-life friction or inertia in Minecraft!

File: mappings/net/minecraft/entity/LivingEntity.mapping
Comment (liach) [inline mappings/net/minecraft/entity/LivingEntity.mapping]:
```suggestion
	METHOD method_35053 hasNoDrag ()Z
	METHOD method_35054 setNoDrag (Z)V
		ARG 1 noDrag
```

**Review by liach — APPROVED** [review]

**Review by enbrain — APPROVED** [review]

**Pull request merged**
